### PR TITLE
Complete testing and fix some issues as below

### PR DESCRIPTION
**Pull request by krnick**
1. complete testing of most programs
2. fixes some features (the rule checker using same instance lead to detection error in main)
3. string optimization and refactoring
4. follow the coding style based on docs

### CODE REVIEW

File: quark/Evaluator/pyeval.py [hunk 1]
```diff
@@ -33,20 +30,20 @@ def __invoke(self, instruction):
         """
 
         executed_fuc = instruction[-1]
-        reg_list = instruction[1: len(instruction) - 1]
+        reg_list = instruction[1 : len(instruction) - 1]
```
Comment (sidra-asa) [inline quark/Evaluator/pyeval.py]: It seems not to fit PEP8.

File: quark/Evaluator/pyeval.py [hunk 2]
```diff
@@ -55,8 +52,7 @@ def __invoke(self, instruction):
                     executed_fuc + "(" + ",".join(value_of_reg_list) + ")"
                 )
         # push the return value into ret_stack
-        self.ret_stack.append(
-            executed_fuc + "(" + ",".join(value_of_reg_list) + ")")
+        self.ret_stack.append(executed_fuc + "(" + ",".join(value_of_reg_list) + ")")
```
Comment (sidra-asa) [inline quark/Evaluator/pyeval.py]: How about this:
`self.ret_stack.append(f"{executed_fuc}({",".join(value_of_reg_list)})}")`

Comment (krnick) [reply quark/Evaluator/pyeval.py]: LGTM

**Comment by krnick** [comment]
done this part of the f-string, if there are other places, then use refactor branch to refactor next time.

**Comment by sidra-asa** [comment]
It looks good to me.

**Pull request merged**